### ISSUE TITLE
aegis does not pick up AWS environmental variables

### ISSUE THREAD

**Issue by pbnj** [nortonlifelock/aegis#19]
I have AWS API keys set as environmental variables and I can verify that I can generate AWS KMS data keys via: `$ aws kms generate-data-keys --key-id "xxx" --number-of-bytes 1024`

However, `aegis init` crashes with:
```
AccessDeniedException: User: arn:aws:iam::xxxxxxxxxx:user/foo is not authorized to perform: kms:GenerateDataKey on resource: arn:aws:kms:us-west-2:xxxxxxxxxx:key/exxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx with an explicit deny
        status code: 400, request id: 30b816c5-42ae-4ea1-afe1-4f3e996a5a1b
```

Presuming because of [this line](https://github.com/nortonlifelock/aegis/blob/326b68251bf9edf16b9b20e1e19fe5404ff0a030/setup/install-config/install-config.go#L90) which should be replaced with [`crypto.CreateKMSClient`](https://github.com/nortonlifelock/crypto/blob/master/kms.go#L68)

**Comment by pbnj** [comment]
To clarify, the reason why I get `AccessDeniedException` is because my AWS IAM user account has bare-minimum permissions (enough to generate temporary security tokens via AWS STS + MFA or to assume an IAM role, which is a security best-practice to limit blast-radius of compromised individual IAM user accounts)

**Comment by ryan-everhart** [comment]
The error is being produced by the [`Encrypt method`](https://github.com/nortonlifelock/aegis/blob/326b68251bf9edf16b9b20e1e19fe5404ff0a030/setup/install-config/install-config.go#L93) which is where the GenerateKeyData permission is required. If your account does not have access to that permission, it will not be able to Encrypt with KMS while the application is running.

I will investigate how to use KMS without requiring the GenerateKeyData permission

**Comment by ryan-everhart** [comment]
@pbnj I removed the dependency on the GenerateKeyData permission. If you pull the latest commit you should only require the KMS Encrypt/Decrypt permissions

**Comment by pbnj** [comment]
@ryan-everhart  - I don't think the issue is with the particular KMS method.
The issue is that `aegis` is not using the correct AWS credentials. 

I have 2 sets of credentials:
- `$HOME/.aws/credentials` are my user credentials, which has no permissions except to call AWS STS or to assume an IAM role.
- After I call AWS STS, I set those AWS ACCESS KEY and AWS SECRET KEY tokens as AWS environment variables.

The AWS CLI and most other AWS tools I have used (e.g. terraform aws provider) follow a precedence (for more details, see the [docs](https://docs.aws.amazon.com/cli/latest/topic/config-vars.html)).

Aegis doesn't seem to follow this precedence and it directly grabs credentials from the local credential file [here](https://github.com/nortonlifelock/aegis/blob/42c5a35db3bdbc479b8d861e30b5835fefaabd1e/internal/setup/install-org/install-org.go#L403).

**Comment by ryan-everhart** [comment]
@pbnj I confirmed verified the behavior of CreateKMSClient first checking for environment variables. The install script now calls that method if environment variables are set

**Comment by pbnj** [comment]
Pulled latest aegis and tried to initialize config and org. 
Config initialized successfully, but `-init-org` crashes with the following error:
```
Encrypting Organization encryption key using KMS...
Done
Generating secure Organization encryption key using crypto/rand
AccessDeniedException: User: arn:aws:iam::[REDACTED]:user/pbenjamin is not authorized to perform: kms:Encrypt on resource: arn:aws:kms:us-west-2:[REDACTED]:key/[REDACTED] with an explicit deny
        status code: 400, request id: 72e86eb4-2d13-4aa4-bded-c6411c4f54f7
```

Seems that [this](https://github.com/nortonlifelock/aegis/blob/master/internal/setup/install-org/install-org.go#L50) is the offending line because it uses `default` profile in my `~/.aws/credentials` and not the `AWS_*` environment variables.

**Comment by ryan-everhart** [comment]
@pbnj Good catch - I didn't catch that method call in the organization initialization. If you pull form master, you should find that the organization initialization now checks for the environment variables.

Just a note: this method is only called during the initialization process, and in a separate tool (kms_tool.go). Aegis itself does not use crypto.CreateKMSClientWithProfile during execution.
